road/stats.py

```diff
@@ -0,0 +1,306 @@
+        count_df = pd.concat([count_df, total_row], ignore_index=True)
+        count_df = reorder_dataframe(count_df, group_var)
+
+        # Round the dataframes for better readability
+        percent_rmse_df = percent_rmse_df.round(1)
+        relative_error_df = relative_error_df.round(5)
+        est_obs_ratio_df = est_obs_ratio_df.round(3)
+
+        # Save the dataframes to CSV
+        group_var_no_spaces = group_var.replace(" ", "").lower()
+        file_prefix = f"{group_var_no_spaces}_"
+        count_df.to_csv(f'{file_prefix}count.csv', index=False)
+        percent_rmse_df.to_csv(f'percent_rmse_{group_var}.csv')
+        relative_error_df.to_csv(f'relative_error_{group_var}.csv')
+        est_obs_ratio_df.to_csv(f'est_obs_ratio_{group_var}.csv')
+
+        # Melt dataframes for easier plotting or analysis
+        melted_percent_rmse_df = percent_rmse_df.melt(
+            id_vars=[group_var], var_name='Time Period', value_name='Percent RMSE')
+        melted_relative_error_df = relative_error_df.melt(
+            id_vars=[group_var], var_name='Time Period', value_name='Relative Error')
+        melted_est_obs_ratio_df = est_obs_ratio_df.melt(
+            id_vars=[group_var], var_name='Time Period', value_name='Est/Obs Ratio')
+
+        # Save melted dataframes to CSV
+        melted_percent_rmse_df.to_csv(
+            f'{file_prefix}percent_rmse_melted.csv', index=False)
+        melted_relative_error_df.to_csv(
+            f'{file_prefix}relative_error_melted.csv', index=False)
+        melted_est_obs_ratio_df.to_csv(
+            f'{file_prefix}est_obs_ratio_melted.csv', index=False)
+
+        # generate the vega-lite files
+        generate_and_save_vega_lite_configs(group_var, file_prefix)
+
+
+def generate_and_save_vega_lite_configs(group_var, file_prefix):
+    output_dir = os.getcwd()
+    os.makedirs(output_dir, exist_ok=True)
+
+    metrics = ["percent_rmse", "relative_error", "est_obs_ratio"]
+    metric_fields = ["Percent RMSE", "Relative Error", "Est/Obs Ratio"]
+    file_suffixes = [
+        "percent_rmse_melted.csv",
+        "relative_error_melted.csv",
+        "est_obs_ratio_melted.csv"]
+
+    group_var_sort_orders = {
+        'AT Group': [
+            'Core/CBD',
+            'UrbBiz',
+            'Urb',
+            'Sub',
+            'All Locations'],
+        'FT Group': [
+            'Fwy/Ramp',
+            'Art',
+            'Col',
+            'Loc',
+            'All Locations'],
+        'Observed Volume': [
+            '<10k',
+            '10-20k',
+            '20-50k',
+            '>=50k',
+            'All Locations'],
+    }
+    custom_order = group_var_sort_orders.get(group_var, None)
+
+    for metric, y_field, file_suffix in zip(
+            metrics, metric_fields, file_suffixes):
+        file_path = os.path.join(output_dir, f"{file_prefix}{file_suffix}")
+
+        # Check if file exists
+        if not os.path.exists(file_path):

```

Comment:
I need a little help understanding this: why is the file only created/updated if it already exists?